### DISCUSSION TITLE
Child Order Changes in Instance

### DISCUSSION THREAD

**Discussion by maltman92** [nxt-dev/nxt_editor#184] [Q&A] [answered]
I'm having an issue when I instance a node tree with multiple children that sometimes the child order will end up different in the instance. Does anyone know what could be causing this? I am creating the instance on a layer above the one I am instancing from. Any help is appreciated. Thank you!

-Mike

**Comment by ImLucasBrown** [comment, marked as answer]
Hi Mike, sorry to hear you're having issues.
I'm actually working on some child order fixes right now and any additional information you can provide would be super valuable.
Can you provide steps to reproduce or a copy of the broken graph(s)?

**Reply by maltman92** [reply]
No worries. I've been having a lot of fun using NXT overall. Thank you for all the work you and your team have put into it. Here is a save out of my rig_components layer. I tried to strip it down to just what's causing the problem. It has a '_component' which 'limb' is an instance of and then 'arm' and 'leg'. In this example the 'init' and 'build' blocks get flopped for the leg component and the order under 'init' and 'init/rigGroups' get messed up as well. If you mute 'maya_utils' the order under 'init' corrects itself but 'init' and 'build' stay in the wrong order and 'init/rigGroups' also.

Thank you for taking the time to look at this! Looking forward to see what you find!

Best,
Mike
[childOrderExample.zip](https://github.com/nxt-dev/nxt_editor/files/6297671/childOrderExample.zip)

**Reply by ImLucasBrown** [reply]
Hmm this is an interesting bug for sure. Its related to relative instancing. I'll keep looking into it and hopefully will have a solution in the next release.
In the meantime you can get around this bug by splitting your `/rig_components` into a few trees. 
Like this: 
![image](https://user-images.githubusercontent.com/54835354/114634010-a0628680-9c8f-11eb-89f5-68797a7cf300.png)
They seem to stay in sync correctly with this change.
![ChildOrder01](https://user-images.githubusercontent.com/54835354/114634123-e4ee2200-9c8f-11eb-9fef-c8eacc661070.gif)

---

I've opened an issue to track this https://github.com/nxt-dev/nxt/issues/78

**Reply by maltman92** [reply]
Thank you for this fix! I'll be instancing nodes like this moving forward and subscribed to follow the issue you attached.

Best,
Mike
